GoRogue/MapViews/IMapViewExtensions.cs

```diff
@@ -337,5 +337,25 @@ public static Point RandomPosition<T>(this IMapView<T> mapView, IGenerator? rng
 
             return new Point(rng.Next(mapView.Width), rng.Next(mapView.Height));
         }
+
+        /// <summary>
+        /// Converts this IMapView<T> into an array of T[]
+        /// </summary>
+        /// <param name="self">the map view being transformed</param>
+        /// <typeparam name="T">The type of which to return an array</typeparam>
+        /// <returns></returns>
+        public static T[] ToArray<T>(this IMapView<T> self)
+        {
+            List<T> items = new List<T>();
+            for (int y = 0; y < self.Height; y++)
+            {
+                for (int x = 0; x < self.Width; x++)
+                {
+                    items.Add(self[x, y]);
+                }
+            }
+
+            return items.ToArray();
```

Comment:
I think it's more efficient to replace this with something that avoids 2x copies (item to list, then list to array):
```suggestion
            var array = new ArrayMap<T>(self.Width, self.Height);
            array.ApplyOverlay(self);
            
            return array; // Rely on implicit conversion between ArrayMap<T> and T[]
```

Reply:
First off, great suggestion and I'll add it to the next commit.

Use case is this: I'm trying to get map data to render in SadConsole, and SadConsole needs an array of `ColoredGlyph` to render. I thought that if I was doing it for the integration, then it might be something other people might use (for SadConsole, or for other rendering solutions). 

If you think this is too specific of a use case, it could go in a different project